app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -47,26 +47,27 @@ export class ProfileForm extends Component {
     }
   }
 
-  onSubmit() {
-    const csrfToken = getContentOfToken('csrf-token');
+  async onSubmit() {
     const { formValues, last_onboarding_page } = this.state;
-    fetch('/onboarding_update', {
-      method: 'PATCH',
-      headers: {
-        'X-CSRF-Token': csrfToken,
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        user: { last_onboarding_page },
-        profile: { ...formValues },
-      }),
-      credentials: 'same-origin',
-    }).then((response) => {
-      if (response.ok) {
-        const { next } = this.props;
-        next();
+    const { username, ...newFormValues } = formValues;
+    try {
+      const response = await request('/onboarding_update', {
+        method: 'PATCH',
+        body: {
+          user: { last_onboarding_page, username },
+          profile: { ...newFormValues },
+        },
+      });
+      if (!response.ok) {
+        throw response;
```

Comment:
Refactor (blocking): Provide feedback to the user if there is an issue with the response. Throwing the response here only log to the browser's console and there is no feedback to the user. Opt to set the `errorMessage` state property with a friendly generic message, e.g. "Unable to continue, an error occurred. Please try again." or something along those lines. I imagine we have wording for a message like this elsewhere in the application.

Note, to test this, I changed the request to give a 404.

<img width="1815" alt="image" src="https://user-images.githubusercontent.com/833231/108648356-ffbeca00-7488-11eb-8e97-ee3bfcaec007.png">

Reply:
@nickytonline Addressed this in the last commit! The only error I'm sending custom messages for is 422, so anything else should be handled in this more generic way for the time being:
![image](https://user-images.githubusercontent.com/11466782/108892802-ed05db80-75d5-11eb-8ade-c180a327291b.png)

Good call on the Honeybadger logging - totally forgot to include that. :smile:

Reply:
I see the generic error visually now. 😎

refactor (blocking): Error appears saying username can't be blank when you click continue even if the username is populated.

![image](https://user-images.githubusercontent.com/833231/108901963-825e9b00-75e9-11eb-94f8-e7f49c574d80.png)

The issue appears to be that the username is not in the component state yet. It gets set when you change the username field.

![image](https://user-images.githubusercontent.com/833231/108902256-e2edd800-75e9-11eb-8738-ef0808f92cae.png)

You can set up the initial state like so in the constructor to resolve the issue.

```javascript
    this.state = {
      groups: [],
      formValues: { username: this.user.username },
      canSkip: false,
      last_onboarding_page: 'v2: personal info form',
    };
```